java/com/google/flatbuffers/Reflection.java

```diff
@@ -0,0 +1,57 @@
+package com.google.flatbuffers;
+
+import com.google.flatbuffers.reflection.BaseType;
+import com.google.flatbuffers.reflection.Field;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Help class to manipulate flatbuffers document by reflection.
+ */
+public final class Reflection {
+  private Reflection() {
+    throw new AssertionError("Not for you!");
+  }
+
+  public static Table getRootTable(ByteBuffer _bb) {
+    return getRootTable(_bb, new Table());
+  }
+
+  public static Table getRootTable(ByteBuffer _bb, Table obj) {
+    _bb.order(ByteOrder.LITTLE_ENDIAN);
+    return obj._init(_bb.getInt(_bb.position()) + _bb.position(), _bb);
+  }
+
+  public static boolean hasValue(Table table, Field field) {
+    int o = table.__offset(field.offset());
+    return o != 0;
+  }
+
+  public static short getShortField(Table table, Field field) {
+    return getShortField(table, field, (short) field.defaultInteger());
+  }
+
+  public static short getShortField(Table table, Field field, short defaultValue) {
+    field.type().baseType();
```

Comment:
Done. Sorry just want to have answer on the name of the method for now I didn't check the code enough.
Sorry for that.